component/loki/source/cloudflare/cloudflare.go

```diff
@@ -73,10 +75,15 @@ func (c *Arguments) Validate() error {
 	if c.PullRange < 0 {
 		return fmt.Errorf("pull_range must be a positive duration")
 	}
-	_, err := cft.Fields(cft.FieldsType(c.FieldsType))
+	_, err := cft.Fields(cft.FieldsType(c.FieldsType), c.AdditionalFields)
 	if err != nil {
-		return fmt.Errorf("invalid fields_type set; the available values are 'default', 'minimal', 'extended' and 'all'")
+		return fmt.Errorf("invalid fields_type set; the available values are 'default', 'minimal', 'extended', 'custom' and 'all'")
 	}
+
+	if invalidFields := cft.FindInvalidFields(c.AdditionalFields); len(invalidFields) > 0 {
+		return fmt.Errorf("invalid additional_fields: %v", invalidFields)
+	}
```

Comment:
Thinking it through, it seems like the cloudflare client library doesn't validate fields against any list. I also see in documentation that the available fields list is provided by a live API: https://developers.cloudflare.com/logs/logpull/requesting-logs#fields 

So I think it's most future proof to remove validation here and always allow any custom fields - that way when new fields are added, we don't need to change it.

Reply:
Right, but in this case we would lose consistency with the `allFields`. If you need to pick `allFields` and then add more in the additional_field to select all of them because `allFields` does not actually contains all valid fields, that would be confusing. Maybe one way to solve this would be to rename allFields to something else?

Reply:
all fields is already outdated, e.g. it doesn't have BotDetectionIDs which is a valid field as per their docs: https://developers.cloudflare.com/logs/reference/log-fields/zone/http_requests
The reason I want to remove the validation is that customers can specifically start using these new fields without having to file an issue with Agent. We could rename all, but it would be a breaking change. I think the best we can do here is write in the docs something like this: 

> all fields type includes a snapshot of fields available at the time of writing this. If new fields are made available via Cloudflare API, you can get them by using the additiona_fields argument.